k8s-scheduler/src/test/java/org/dcm/WorkloadGeneratorIT.java

```diff
@@ -270,6 +311,34 @@ void runTrace(final DefaultKubernetesClient client, final String fileName, final
                         final String appName = "app-" + taskCount;
                         pod.getMetadata().setName(appName + "-" + podCount);
 
+                        if (createAffinityRequirements) {
+                            // creating anti-affinity affinity requirements, with In operator
+                            final String [] labels = getLabels(deploymentAffinity, taskCount, podCount);
+                            final LabelSelector labelSelector = new LabelSelector();
+                            final List<String> matchingValues = new ArrayList<>();
+                            matchingValues.add(labels[1]);
+                            final LabelSelectorRequirement requirement =
+                                    new LabelSelectorRequirement(labels[0], "In", matchingValues);
+                            final List<LabelSelectorRequirement> requirements = new ArrayList<>();
+                            requirements.add(requirement);
+                            labelSelector.setMatchExpressions(requirements);
+                            final PodAffinityTerm podAffinityTerm = new PodAffinityTerm();
+                            podAffinityTerm.setLabelSelector(labelSelector);
+                            podAffinityTerm.setTopologyKey("key");
```

Comment:
Was the podspec generated here tested against a k8s cluster (say, one using kind?) ?  `setTopologyKey("key")` should not be allowed, it isn't a valid topology key.

I think a much easier alternative would be to add a pod-resource file that already has the right PodAntiAffinity pod-spec, but in the code here, simply edit the label key/values (like we're doing for the pod name and cpu/mem requests). That would make this code far more concise.

You can test this by bringing up a k8s cluster locally on your laptop using kind.